.github/workflows/check.yaml

```diff
@@ -26,16 +26,47 @@ jobs:
       working-directory: .
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
     needs: lint-unit
+    name: functional tests
+    runs-on: ${{ matrix.runs-on }}
+    timeout-minutes: 120
     strategy:
       fail-fast: false
       matrix:
-        command: ['TEST_JUJU3=1 make functional']  # TEST_JUJU3 needed due https://github.com/openstack-charmers/zaza/blob/b22c2eed4c322f1dfc14ffb2d31e0dd18c911a40/setup.py#L47 to support Juju3+
-    with:
-      command: ${{ matrix.command }}
-      juju-channel: "3.4/stable"
-      nested-containers: false
-      provider: "lxd"
-      python-version: "3.10"
-      timeout-minutes: 120
+        juju-channel: ["3.4/stable"]
+        runs-on:
+          - ['ubuntu-latest']  # github hosted runners are amd64
+          - ['Ubuntu_ARM64_4C_16G_01']
+    steps:
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Setup Juju environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: "lxd"
+          juju-channel: ${{ matrix.juju-channel }}
+          charmcraft-channel: "2.x/stable"
+
+      - name: Show juju information
+        run: |
+          juju version
+          juju controllers | grep Version -A 1 | awk '{print $9}'
```

Comment:
In the past we have some issue that controller version is different than snap juju version/python-libjuju. This step just make debugging more easier.

Reply:
Thanks @aieri :)  Maybe we can look into improving this debug output when we add more debugging info (centrally, from the automation repo).

Reply:
> some issue that controller version is different than snap juju version/python-libjuju

@jneo8 was this only for the external controller?  It seems odd that it would affect here, where juju bootstraps the controller locally.

Reply:
I remember there weer some issues that specific python-libjuju version not work with specific juju controller version.

Reply:
Sounds good - yeah I haven't heard about any issues for a while, so I guess it's not a problem any more.

Ideally we'll add a final debugging step on failure - print this juju information, juju status, juju debug-log, etc.